java/code/src/com/suse/manager/model/maintenance/CalendarFactory.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2020 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package com.suse.manager.model.maintenance;
+
+import com.redhat.rhn.common.hibernate.HibernateFactory;
+import com.redhat.rhn.domain.user.User;
+
+import org.apache.log4j.Logger;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * {@link HibernateFactory} for {@link MaintenanceCalendar}
+ */
+public class CalendarFactory extends HibernateFactory {
+
+    private static Logger log = Logger.getLogger(CalendarFactory.class);
+
+    /**
+     * Save a MaintenanceCalendar
+     * @param calendar the calendar
+     */
+    public void save(MaintenanceCalendar calendar) {
+        getSession().save(calendar);
+    }
+
+    /**
+     * Remove a {@link MaintenanceCalendar}
+     * @param calendar to remove
+     */
+    public void remove(MaintenanceCalendar calendar) {
+        removeObject(calendar);
+    }
+
+    /**
+     * List Maintenance Calendar Labels belonging to the given User
+     * @param user the user
+     * @return a list of Calendar labels
+     */
+    public List<String> listCalendarLabelsByUser(User user) {
```

Comment:
```suggestion
    public List<String> listLabelsByUser(User user) {
```
Or was `Calendar` left here for a reason?

Reply:
Yes, I left this one on purpose. I was thinking, that `listLabelsByUser` was too puristic :) Same for the `Schedule` case...